lib/vtls/openssl.c

```diff
@@ -3020,6 +3021,17 @@ static CURLcode ossl_connect_step1(struct connectdata *conn, int sockindex)
   }
 #endif
 
+  if (ssl_cainfo_blob) {
+    X509 *cert;
+    BIO *ssl_cainfo_bio = BIO_new_mem_buf(ssl_cainfo_blob->data, (int)ssl_cainfo_blob->len);
+    if(!ssl_cainfo_bio)
+        return CURLE_SSL_CERTPROBLEM;
+    while (cert = PEM_read_bio_X509(ssl_cainfo_bio, NULL, 0, NULL)) {
+      X509_STORE_add_cert(SSL_CTX_get_cert_store(backend->ctx), cert);
```

Comment:
It adds the cert to the store, right? Will repeated invokes re-using the same handle then readd the same cert again?

Reply:
Judging from the code, this is done absolutely the same way and at the same point as importing the windows cert store, as well as loading certs from files/dirs. The SSL context is created afresh at the start of this function:
`  if(backend->ctx)
    SSL_CTX_free(backend->ctx);
  backend->ctx = SSL_CTX_new(req_method);
`